### PR TITLE
feat(components): add `PasswordField` 👀 

### PR DESCRIPTION
**Pull request by divyanshu013**
## API design

API design is quite similar to `TextInput`. Check the [formatted version](https://github.com/razorpay/blade/blob/div/password-field/packages/blade/src/components/Input/PasswordField/_decisions/decisions.md).

## Details

Adds the base functionality of `PasswordField` and the API decisions in this PR. Tests and exposing the component remains, will be creating a separate PR to keep this one small and easier for reviews.

**Breakdown**

This PR:
- [x] Web implementation
- [x] API decisions
- [x] Native implementation

Next PRs:
- [x] Add tests #715 
- [x] Expose component #716

### Android

| Error state | Success state |
| - | - |
| Screenshot_1663321556 | Screenshot_1663321672 |

### iOS

| Error state | Max characters |
| - | - |
| Simulator Screen Shot - iPhone 13 - 2022-09-16 at 15 56 15 | Simulator Screen Shot - iPhone 13 - 2022-09-16 at 15 57 10 |

### Web

Tested on Safari:

| Suggest password | Use password |
| - | - |
| Screen Shot 2022-09-16 at 16 07 13 | Screen Shot 2022-09-16 at 16 08 07 |

### CODE REVIEW

File: packages/blade/src/components/Input/PasswordField/PasswordField.tsx [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+   * @default true
+   */
+  showRevealButton?: boolean;
+
+  /**
+   * Displays asterisk (`*`) when `isRequired` is enabled
+   *
+   * @default none
+   */
+  necessityIndicator?: Exclude<BaseInputProps['necessityIndicator'], 'optional'>;
```
Comment (anuraghazra) [inline packages/blade/src/components/Input/PasswordField/PasswordField.tsx:25]: 😄

File: packages/blade/src/components/Input/PasswordField/PasswordField.tsx [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+  const revealButton = showRevealButton ? (
+    <IconButton
+      size="medium"
+      icon={revealButtonIcon}
+      onClick={toggleIsRevealed}
+      accessibilityLabel={accessibilityLabel}
+    />
+  ) : null;
+
+  const trailingFooterSlot = (value: Maybe<string>): Maybe<ReactNode> =>
```
Comment (anuraghazra) [inline packages/blade/src/components/Input/PasswordField/PasswordField.tsx:113]: Don't need maybe type. we can remove it.

```ts
  const trailingFooterSlot = (value?: string): ReactNode =>
```

File: packages/blade/src/components/Input/PasswordField/PasswordField.tsx (hunk 1 above)
Comment (divyanshu013) [reply packages/blade/src/components/Input/PasswordField/PasswordField.tsx:25]: Learning from the guru 😹

File: packages/blade/src/components/Input/PasswordField/PasswordField.tsx [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+    maxCharacters ? (
+      <Box marginTop="spacing.2" marginRight="spacing.1">
+        <CharacterCounter currentCount={value?.length ?? 0} maxCount={maxCharacters} />
+      </Box>
+    ) : null;
+
+  return (
+    <BaseInput
+      id={id}
```
Comment (anuraghazra) [inline packages/blade/src/components/Input/PasswordField/PasswordField.tsx:122]: I think this ID is not meant to be passed by users 

ref: 
https://razorpay.slack.com/archives/G01B3LQ9H0W/p1663568715561059

File: packages/blade/src/components/Input/PasswordField/PasswordField.stories.tsx [hunk 4]
```diff
@@ -0,0 +1,220 @@
...
+
+export const AutoComplete = PasswordFieldTemplate.bind({});
+AutoComplete.args = {
+  autoCompleteSuggestionType: 'newPassword',
+};
+AutoComplete.parameters = {
+  docs: {
+    description: {
+      story:
+        '`autoCompleteSuggestionType` can be used to tell the platform if the input field is being used for inputting new password or current password. This provides hints to browser autofill and password managers. **Note:** there is a [known issue](https://github.com/facebook/react-native/issues/21911) for iOS.',
```
Comment (anuraghazra) [inline packages/blade/src/components/Input/PasswordField/PasswordField.stories.tsx:111]: nice

**Review by chaitanyadeorukhkar — APPROVED** [review]

File: packages/blade/src/components/Input/BaseInput/StyledBaseInput.native.tsx [hunk 5]
```diff
@@ -136,6 +137,7 @@ export const StyledBaseInput = React.forwardRef<TextInput, StyledBaseInputProps>
               ] as StyledComponentAutoCompleteAndroid)
             : undefined
         }
+        secureTextEntry={type === 'password'}
```
[the remark this answers is not in the document]
Comment (divyanshu013) [reply packages/blade/src/components/Input/BaseInput/StyledBaseInput.native.tsx]: Good idea, done in https://github.com/razorpay/blade/pull/712/commits/1dd8d8eaa42f207b4fd5254a5561ee6d9b35bf2d

File: packages/blade/src/components/Input/BaseInput/_decisions/_decisions.md [hunk 6]
```diff
@@ -20,7 +20,7 @@ The BaseInput is the underlying input field that will in turn be used to render
...
+| type                                                                                                       | `text`, `password`, `search`, `telephone`, `email`, `url`, `numeric`, `password`                                                                                                                                                                       | No       | `text`                                | type of text to be rendered. this will also determine the keyboard type to be show on mobile devices                                                                                                                                                                                  |
```
[the remark this answers is not in the document]
Comment (divyanshu013) [reply packages/blade/src/components/Input/BaseInput/_decisions/_decisions.md]: Oops, fixed in https://github.com/razorpay/blade/pull/712/commits/1dd8d8eaa42f207b4fd5254a5561ee6d9b35bf2d

File: packages/blade/src/components/Input/BaseInput/StyledBaseInput.native.tsx (hunk 5 above)
[the remark this answers is not in the document]
Comment (divyanshu013) [reply packages/blade/src/components/Input/BaseInput/StyledBaseInput.native.tsx]: Need to revert this, it seems like `secureTextEntry` doesn't work with `undefined`

**Pull request merged**